docs_src/design/adr/device-service/0021-Profile-Changes.md

```diff
@@ -0,0 +1,126 @@
+| Name | string | Never | This required property must be unique to the EdgeX instance and cannot be changed.  The profile name can never be empty and must be unique across the entire EdgeX instance. |
+| Manufacturer | string | Anytime | As this property does not affect any functionality other than profile query, this property can be change at any time – to include setting the string to an empty string. |
+| Description | string | Anytime | As this property does not affect any functionality other than profile query, this property can be change at any time – to include setting the string to an empty string. |
+| Model | string | Anytime | As this property does not affect any functionality other than profile query, this property can be change at any time – to include setting the string to an empty string. |
+| Labels | string array | Anytime | As this property does not affect any functionality other than profile query, this property can be change at any time – to include adding, removing or modifying any of the strings to the array of labels. |
+| Device Resources | array | Situational | When the device profile is not associated to any device, reading or event, the device resource list can be modified (i.e., adding or removing the associated device resources).  Once the profile is associated to a device, reading or event, only new resources can be added to the array.  Deleting is prohibited.|
+| Device Commands | array | Situational | When the device profile is not associated to any device, reading or event, the device commands list can be modified (i.e., adding or removing the associated device resources).  Once the profile is associated to a device, reading or event, only new commands can be added to the array.  Deleting is prohibited.  Device commands are optional and so the array can always be empty. |
+| *Device Resource Properties* |||
+| Description | string | Anytime | As this property does not affect any functionality, this property can be change at any time – to include setting the string to an empty string. |
+| Name | string | Situational | When the device profile is not associated to any device, reading or event, this field can be modified.  It can never be blank and must be unique across the entire profile. |
+| IsHidden | bool | Anytime | This field allows the device resource to be exposed (default) via the command service.  This change does not affect the behavior but only visibility.  This field is optional but is false by default when not provided. |
+| Tag | string | Situational | As this property does not affect any functionality, this property can be change at any time – to include setting the string to an empty string. |
+| Properties | Resource Properties | Situational | When the device profile is not associated to any device, reading or event, this field can be modified.  Because the ValueType and ReadWrite properties of this field are required, it can never be empty. |
```

Comment:
Shouldn't we allow modification of `ReadWrite` property at any time? Changing this doesn't change the definition of the resource, it just changes what operations area allowed to be performed on this resource.